runtime/ceylon/language/Integer.java

```diff
@@ -420,7 +421,7 @@ public double getFloat() {
     public static double getFloat(long value) {
         if (value >= 9007199254740992L
                 || value <= -9007199254740992L) {
-            throw new OverflowException();
+            throw new OverflowException(value + " cannot be coerced into a 64 bit floating point value");
```

Comment:
> Or we could simply ignore any lost precision here. 

That's a pretty good question. I'm inclined to think we probably need two of these. `float` is pretty good when you need conversion to `Float` without loss of precision. But surely there are times when you need to do the conversion even allowing for some loss of precision. Currently we simply don't support that. So I think we should add a `closestFloat` or `approximateFloat` or something like that.

Do you want to write up a feature req for that?